plutus-contract/src/Wallet/API.hs

```diff
@@ -91,12 +95,20 @@ payToPaymentPublicKeyHash range v pk = do
     utx <- either (throwError . PaymentMkTxError)
                   pure
                   (Constraints.mkTx @Void mempty constraints)
-    let adjustedUtx = Constraints.adjustUnbalancedTx utx
+    let adjustedUtx = adjustUnbalancedTx utx
     unless (utx == adjustedUtx) $
       logWarn @Text $ "Wallet.API.payToPublicKeyHash: "
                    <> "Adjusted a transaction output value which has less than the minimum amount of Ada."
     balancedTx <- balanceTx adjustedUtx
     either throwError signTxAndSubmit balancedTx
+    where
+        adjustUnbalancedTx :: Constraints.UnbalancedTx -> Constraints.UnbalancedTx
+        adjustUnbalancedTx = over (tx . Tx.outputs . traverse) adjustTxOut
+          where
+            adjustTxOut :: Tx.TxOut -> Tx.TxOut
+            adjustTxOut txOut =
+              let missingLovelace = max 0 (minAdaTxOut - Ada.fromValue (Tx.txOutValue txOut))
+               in txOut { Tx.txOutValue = Tx.txOutValue txOut <> Ada.toValue missingLovelace }
```

Comment:
Good question.

In general `minAda` is not enough but it seems that `payToPaymentPublicKeyHash` is being used only in tests, use cases, and on PAB's start.

We can bring protocol params as dependency here but it will require us to fix all tests to pass it. Or we can use the default protocol parameters as in the emulator handler.

Reply:
Since the PAB is actually using it through `payToPaymentPublicKey` in `Core`, I would bring the protocol params as dependency and update the tests. Maybe in another PR? Up to you though.